src/main/scala/com/amazon/deequ/profiles/ColumnProfiler.scala

```diff
@@ -347,26 +353,33 @@ object ColumnProfiler {
   private[this] def extractGenericStatistics(
       columns: Seq[String],
       schema: StructType,
-      results: AnalyzerContext)
+      results: AnalyzerContext,
+      predefinedTypes: Map[String, DataTypeInstances.Value] = Map.empty)
     : GenericColumnStatistics = {
 
     val numRecords = results.metricMap
       .collect { case (_: Size, metric: DoubleMetric) => metric.value.get }
       .head
       .toLong
 
+
     val inferredTypes = results.metricMap
+      .filterNot{
+        case (analyzer: DataType, _) => predefinedTypes.contains(analyzer.column)
+        case _ => true}
       .collect { case (analyzer: DataType, metric: HistogramMetric) =>
-        val typeHistogram = metric.value.get
-        analyzer.column -> DataTypeHistogram.determineType(typeHistogram)
+          val typeHistogram = metric.value.get
+          analyzer.column -> DataTypeHistogram.determineType(typeHistogram)
       }
 
     val typeDetectionHistograms = results.metricMap
+      .filterNot{
+        case (analyzer: DataType, _) => predefinedTypes.contains(analyzer.column)
+        case _ => true}
```

Comment:
move the bracket to the next line, please

Reply:
Sure thing!